clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java

```diff
@@ -718,7 +718,8 @@ private void initializeIsolatedIngestionServer() {
         sslFactory,
         null,
         null,
-        null);
+        null,
+        Optional.of(d2Client)); // TODO: Figure out a way to get D2Client from davinci config to figure out D2Client.
```

Comment:
I'm a bit confused about this TODO here. We have everything we need to resolve this right?